### PR TITLE
Add Travis and Appveyor

### PR DESCRIPTION
**Pull request by JackDunnNZ**

### CODE REVIEW

**Comment by mlubin** [comment]
:+1:

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+language: julia
+os:
+  - linux
+  - osx
+julia:
+  - 0.4
```
Comment (mlubin) [inline .travis.yml]: Should keep testing with 0.3 as long as the package continues to support it

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+sudo: false
+cache:
+  directories:
+  - $HOME/usr
+addons:
+  apt_packages:
+  - gfortran
+env:
+  - MAKEFLAGS="-j3"
+    LD_LIBRARY_PATH=$HOME/usr/lib
```
Comment (mlubin) [inline .travis.yml]: Why is this needed?

Comment (JackDunnNZ) [inline .travis.yml]: I copied from CoinOptServices.jl and didn't try without it. I presume it's so the packages can find the cached libraries and not bulld them again? cc @tkelman

File: .travis.yml (hunk 1 above)
Comment (JackDunnNZ) [inline .travis.yml]: Sure

**Comment by mlubin** [comment]
Looks like there are a couple ``Int32``/``Int64`` mismatches that showed up on win32

**Comment by JackDunnNZ** [comment]
Fixing those now.

File: .travis.yml (hunk 2 above)
Comment (tkelman) [inline .travis.yml]: yeah it's for caching of the built binaries
